File: 5. Longest Palindromic Substring/longest_palindromic_substring.cpp
#include<iostream>
#include<string>

using namespace std;

bool is_palindrome (string sub){
	int len_sub = sub.size();
	int mir = len_sub / 2;
	bool out = false;
	
	for (int i = 0; i < mir; i++){
		if (i == 0){
			out = (sub[i] == sub[len_sub - 1 - i]);
		} else {
			out = out && (sub[i] == sub[len_sub - 1 - i]);
		}
	}
	
	return (out);
}

string longest_substring (string text){
	int len_text = text.size();
	string longest_text = "";
	
	for (int i = 0; i < len_text; i++){
		for (int j = (len_text - i); j >= 0; j--){
			string substring = text.substr(i, j);
			
			if (is_palindrome(substring)){
				if (longest_text.size() < substring.size()){
					longest_text = substring;
					break;
				}
			}
		}
	}
	
	return (longest_text);
}

int main (){
	string word = "bbob";
	cout << "This is the handy work of C++.\n\n\n";
	cout << longest_substring(word);
	cout << "\n\n\n";
	
	return (0);
}

File: 5. Longest Palindromic Substring/longest_palindromic_substring.js
function isPalindrome (sub){
	let lenSub = sub.length;
	let mirror = (lenSub / 2) - ((lenSub / 2) % 1);
	let out = false;
	
	for (let a = 0; a < mirror; a++){
		if (a == 0){
			out = (sub[a] == sub[lenSub - 1 - a]);
		} else {
			out = out && (sub[a] == sub[lenSub - 1 - a]);
		}
	}
	
	return (out);
}

function longestSubString(text){
	let lenText = text.length;
	let longestText = "";
	
	for (let a = 0; a < lenText; a++){
		for (let b = 1; b < lenText; b++){
			let subString = text.substring(a, a + b + 1);
			
			if (isPalindrome(subString)){
				if (longestText.length < subString.length){
					longestText = subString;
				}
			}
		}
	}
	return (longestText);
}

let word = "babad";

console.log(longestSubString(word));
console.log("\n\n\n\n");
